src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerServerEndpoint.java

```diff
@@ -78,12 +82,21 @@ public DockerServerEndpoint(String uri, String credentialsId) {
      *
      * <p>
      * Null to indicate whatever Docker picks by default.
+     * @deprecated use #getDockerHost
      */
     public @Nullable String getUri() {
         return uri;
     }
 
     /**
+     * Gets the endpoint in URI, such as "unix:///var/run/docker.sock".
+     */
+    public @Nonnull String getDockerHost() {
```

Comment:
I also agree with @jtnord that if no systemwide default has been configured, and the endpoint specifies no URI, it would be better to fall back to `null` meaning to not specify `--host` at all and use the default server according to the client on whatever machine it runs on.